src/arch/src/aarch64/regs.rs

```diff
@@ -196,6 +207,236 @@ pub fn read_mpidr(vcpu: &VcpuFd) -> Result<u64> {
     vcpu.get_one_reg(MPIDR_EL1).map_err(Error::GetSysRegister)
 }
 
+/// Get the state of the core registers.
+///
+/// # Arguments
+///
+/// * `vcpu` - Structure for the VCPU that holds the VCPU's fd.
+/// * `state` - Structure for returning the state of the core registers.
+pub fn save_core_registers(vcpu: &VcpuFd, state: &mut kvm_regs) -> Result<()> {
+    let mut off = offset__of!(user_pt_regs, regs);
+    // There are 31 user_pt_regs:
+    // https://elixir.free-electrons.com/linux/v4.14.174/source/arch/arm64/include/uapi/asm/ptrace.h#L72
+    // These actually are the general-purpose registers of the Armv8-a
+    // architecture (i.e x0-x30 if used as a 64bit register or w0-30 when used as a 32bit register).
+    for i in 0..31 {
+        state.regs.regs[i] = vcpu
+            .get_one_reg(arm64_core_reg_id!(KVM_REG_SIZE_U64, off))
```

Comment:
Aren't the id's continuous ? Instead of incrementing the offset, could we start with `id = arm64_core_reg_id!(KVM_REG_SIZE_U64, 0)` and increment the id after processing each register ?